src/cljs/rems/search.cljs

```diff
@@ -7,67 +7,6 @@
             [rems.text :refer [text]]
             [rems.util :refer [fetch focus-when-collapse-opened]]))
 
-(defn reg-fetcher
-  "Registers a set of event handlers and subscriptions for fetching data, with
-  support for search. Throttles automatically so that only one fetch is running
-  at a time.
-
-  Given `(reg-fetcher ::foo \"/url\")`, registers the following:
-
-  - Handler `[::foo]` fetches data from `/url`, i.e. get all items
-  - Handler `[::foo \"bar\"]` fetches data from `/url?query=bar`, i.e. do a search
-  - Subscription `[::foo]` returns the data that was last fetched
-  - Subscription `[::foo :error]` returns the error message if the fetch failed
-  - Subscription `[::foo :initialized?]` returns `true` if the data has been
-    fetched at least once
-  - Subscription `[::foo :fetching?]` returns `true` if a fetch is in progress
-  - Subscription `[::foo :searching?]` returns `true` if a search is in progress,
-    i.e. a fetch after the initial fetch
-
-  All state is stored in `db` under `::foo`, which can be removed to reset state."
-  [id url]
-  (let [result-id (keyword (namespace id)
-                           (str (name id) "-result"))]
-    (rf/reg-event-fx
-     id
-     (fn [{:keys [db]} [_ query]]
-       ;; do only one fetch at a time - will retry after the pending fetch is finished
-       (when-not (get-in db [id :fetching?])
-         (fetch url
-                {:url-params (when query
-                               {:query query})
-                 :handler #(rf/dispatch [result-id {:data %} query])
-                 :error-handler (fn [response]
-                                  ;; We use a custom error reporting mechanism instead of flash-message
-                                  ;; because grabbing the focus is problematic for continuous search.
-                                  (rf/dispatch [result-id
-                                                {:error (flash-message/format-response-error response)}
-                                                query]))}))
-       {:db (-> db
-                (assoc-in [id :query] query)
-                (assoc-in [id :fetching?] true))}))
-
-    (rf/reg-event-db
-     result-id
-     (fn [db [_ result query]]
-       ;; fetch again if the query that just finished was not the latest
-       (let [latest-query (get-in db [id :query])]
-         (when-not (= query latest-query)
-           (rf/dispatch [id latest-query])))
-       (-> db
-           (assoc-in [id :data] (:data result))
-           (assoc-in [id :error] (:error result))
-           (assoc-in [id :initialized?] true)
-           (assoc-in [id :fetching?] false))))
-
-    (rf/reg-sub
-     id
-     (fn [db [_ k]]
-       (case k
-         :searching? (and (get-in db [id :fetching?])
-                          (get-in db [id :initialized?]))
-         (get-in db [id (or k :data)]))))))
```

Comment:
hard to see in the diff: did you just move this to rems.fetcher as-is or edit it?

Reply:
Hmm, I moved it as is and then added some features.